File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function solution(n) {
    
    const pq = [];
    
    while(n > 0){
        var div = n%10;
        pq.push(div);
        n = Math.floor(n/10);
    }
    
    pq.sort((a,b) => a-b);
    
    var answer = 0;
    
    while(pq.length){
        answer += pq.pop();
        if(pq.length) answer *= 10;
    }
    
    return answer;
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;

class Solution {
    public long solution(long n) {
        
        PriorityQueue<Long> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        while(n > 0){
            long div = n%10;
            n /= 10;
            pq.add(div);
        }
        
        long answer = 0;
        
        while(pq.size() > 0){
            answer += pq.poll();
            if(pq.size() > 0) answer *= 10;
        }
        
        return answer;
    }
}